LUSDChickenBonds/src/BLUSDLPZap.sol

```diff
@@ -96,4 +96,54 @@ contract BLUSDLPZap {
 
         return bLUSDLUSD3CRVTokens;
     }
+
+    function removeLiquidityBalanced(uint256 _lpAmount, uint256 _minBLUSD, uint256 _minLUSD) external {
+        // pull LP tokens
+        bLUSDLUSD3CRVLPToken.transferFrom(msg.sender, address(this), _lpAmount);
+
+        // All those balances should be zero, but just in case:
+        uint256 initialLUSD3CRVBalance = lusd3CRVPool.balanceOf(address(this));
+        uint256 initialBLUSDBalance = bLUSDToken.balanceOf(address(this));
+
+        // withdraw bLUSD/LUSD-3CRV
+        bLUSDLUSD3CRVLPToken.approve(address(bLUSDLUSD3CRVPool), _lpAmount);
+        bLUSDLUSD3CRVPool.remove_liquidity(_lpAmount, [_minBLUSD, 0], false, address(this));
+        uint256 bLUSDAmount = bLUSDToken.balanceOf(address(this)) - initialBLUSDBalance;
+        uint256 lusd3CRVAmount = lusd3CRVPool.balanceOf(address(this)) - initialLUSD3CRVBalance;
+
+        // withdraw LUSD from LUSD/3pool, and tranfer it to sender
+        if (lusd3CRVAmount > 0) {
+            lusd3CRVPool.remove_liquidity_one_coin(
+                lusd3CRVAmount,
+                0,
+                _minLUSD,
```

Comment:
...although, I guess this `_minLUSD` implicitly gives us a minimum LUSD3CRV quantity for the above withdrawal. 

Unless the bLUSD-LUSD3CRV pool is so imbalanced that we withdraw 0 LUSD3CRV in L110 (in which case L119 won't get reached), which seems unlikely/impossible.  So almost certainly not an issue.

Reply:
Yes, that was the idea, use the `_minLUSD` to implicitly enforce the other one.
But good point about getting zero LUSD3CRV. I guess it doesn’t hurt to add a check for that.